### PR TITLE
Aderyn Sidebar 

### PR DESCRIPTION
**Pull request by TilakMaddy**
* Introduces Aderyn in the sidebar (random icon from Microsoft github used as of now)
* Allows for scope and exclude input
* Aderyn watch button (when clicked displays message "Aderyn is watching your codebase..."

Todo (maybe in this PR or later)
* *actually* invoke aderyn command 
* persist  the state of chosen values of scope and exlcuded for the project in the user's home directory so they don't have to enter it each time.

### CODE REVIEW

File: src/SidebarProvider.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        case "watch": {
+          vscode.window.showInformationMessage("Aderyn is now watching your codebase ...");
+          const scope = data.value.scope;
+          const exclude = data.value.exclude;
+          // TODO: Run aderyn command here : )
+          console.log({
+            scope, exclude
+          });
+
```
Comment (TilakMaddy) [inline src/SidebarProvider.ts:47]: Here, we're supposed to call aderyn

**Comment by TilakMaddy** [comment]
Here's where I am confused. I need help 

What makes more sense ?

1) aderyn cli takes the responsibility of watching file changes
2) vscode invokes aderyn when there are file changes 

Looks like we're already kind of doing option 2, so should we go with it ?

**Comment by TilakMaddy** [comment]
image

File: media/aderyn.svg
Comment (TilakMaddy) [inline media/aderyn.svg:1]: to replace this with official aderyn svg

**Comment by TilakMaddy** [comment]
Benefits of having a sidebar:
* We can also have sections to show other stuff like attack surface

**Comment by alexroan** [comment]
Hey @TilakMaddy this is not actually doing anything when I hit the button. It says aderyn is watching, but I get no diagnostic feedback at all.

File: src/SidebarProvider.ts (hunk 1 above)
Comment (TilakMaddy) [reply src/SidebarProvider.ts:47]: @alexroan This PR is just a template, here we are supposed to call `runAderyn`

**Comment by TilakMaddy** [comment]
> Hey @TilakMaddy this is not actually doing anything when I hit the button. It says aderyn is watching, but I get no diagnostic feedback at all.

Alex, you should read PR description : )

**Comment by alexroan** [comment]
closing for staleness

**Pull request closed without merging**